render-utils.js

```diff
@@ -0,0 +1,48 @@
+export function renderSongCard(song) {
+    const div = document.createElement('div');
+    const a = document.createElement('a');
+    const img = document.createElement('img');
+    const p = document.createElement('p');
+
+    div.classList.add('song-card');
+    
+    p.textContent = song.title;
+    img.src = `./assets/${song.id}.png`;
+    a.href = `./detail/?id=${song.id}`;
+    //append things to the div
+    div.append(p, img);
+    //append div to dom
+    a.append(div);
+    //return the a
+    return a;
+}
+
+export function renderSongDetail(songData) {
+//create elements
+    const div = document.createElement('div');
+    const img = document.createElement('img');
+    const titleEl = document.createElement('p');
+    const artistEl = document.createElement('p');
+    const albumEl = document.createElement('p');
+    const yearEl = document.createElement('p');
+    const producerEl = document.createElement('p');
+
+    img.src = `./assets/${songData.song.id}.png`;
```

Comment:
here it looks like we're getting an error reading 'song'. maybe it's supposed to be 'songs'? I don't remember the exact shape of your object.